pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptionsUtils.java

```diff
@@ -105,4 +105,56 @@ public static boolean isServerReturnFinalResult(Map<String, String> queryOptions
   public static String getOrderByAlgorithm(Map<String, String> queryOptions) {
     return queryOptions.get(QueryOptionKey.ORDER_BY_ALGORITHM);
   }
+
+  public static int getMultiStageLeafLimit(Map<String, String> queryOptions, int defaultLeafLimit) {
```

Comment:
Curious to know what will be the use of this option to configure leaf limit ?

The other existing options should be enough to control the GROUP BY output / accuracy at leaf layer ?

Reply:
If # of rows scanned exceeds the limit, we may return wrong result right? 

The principle here is set up like limit or cap should be configurable via things other than codes. 
Cluster config is also good but I guess right now, it would be good to start with an option in query and we eventually move some of them to cluster config if we don't need to change it query by query. 

I admit it is kinda hacky to set everything via query option.. but I don't know a better way to make stuff configurable